### DISCUSSION TITLE
React Native Web Support

### DISCUSSION THREAD

**Discussion by ranjithrd** [invertase/react-native-firebase#5506] [General]
Hi, does this package work with [react-native-web (link)](https://necolas.github.io/react-native-web)? Thank you.

**Comment by mikehardy** [comment]
No it does not.
Ideally as we ingest the changes to APIs in firebase-js-sdk v9 it will, but it definitely does not now.
That said, firebase-js-sdk supports react-native-web, since it is pure JS, and you can use tricks like javascript preprocessors to swap in which implementation you use under the firebase-js-sdk APIs in order to use both. I do this to good effect - I used gulp to preprocess before it goes through metro for bundling. ReactXP's 'Todo List" example with their modules is a good example of this.

**Reply by mattgabor** [reply]
@mikehardy can you link the ReactXP "Todo List" example you mentioned? I'm having a hard time finding examples of using @react-native-firebase and the firebase JS SDK in the same app.

**Reply by mikehardy** [reply]
Got what you need right here https://github.com/invertase/react-native-firebase-authentication-example

**Reply by mikehardy** [reply]
(the specific magic is in the "shims" directory in combo with entries referencing those shims in craco.config.js - ReactXP is a dead project, so it is not to be relied on and I just spent a lot of time with the Luna template - base react-native-web+typescript - to replace ReactXP then this auth example template to show firebase working cross-platform, it's the current state of the art and my work projects have been re-platformed on to them)

**Comment by mikehardy** [comment]
react-native-firebase has web support now via "other" platform support which is firebase-js-sdk fallback when not on ios/android:

https://rnfirebase.io/#other--web
https://rnfirebase.io/platforms#other-platforms
